### ISSUE TITLE
ZMQ channel - pushing information about the health of the MISP instance

### ISSUE THREAD

**Issue by adulau** [MISP/MISP#3036]
ZMQ channel - pushing information about the health of the MISP instance like workers or system activity then the misp-dashboard can be used to monitor the instance too.

**Comment by P4rs3R** [comment]
Hi man,
I'm having troubles with ZMQ: "Python ZeroMQ library not installed correctly" - checking the status: "The ZeroMQ server is unreachable."
From Python3 (3.6.3) command line, I've this:
```
zmq.zmq_version() --> 4.1.6
zmq.__version__ --> 17.0.0
```
Latest MISP v2.4.88 (33f9c99205d616b1316841fb8b521ed619f05e61) on your VM (Linux misp 4.13.0-37)
Looked for some info in error log, but nothing interesting found.
PS: I need ZMQ in order to properly use the amazing MISP-Dashboard ;)

**Comment by adulau** [comment]
Could you run this on the VM?  https://github.com/MISP/MISP/blob/2.4/tools/misp-zmq/sub.py to
see if the ZMQ channel works.

**Comment by P4rs3R** [comment]
Restarted the VM. For misp-dashboard, after cloning the git repo (+install dipendencies +edit config), I just run redis-server and (while in dashenv) started zmq_subscriber, zmq_dispatcher and server.py).
I tried `python3 -u /var/www/MISP/tools/misp-zmq/sub.py  | jq`
 (with the argument `-r host` like localhost and external-shown IP), but it shows no data at all.
MISP plugin settings (from web-GUI): all zmq values set to "true". Read also [Koen blog](https://www.vanimpe.eu/2017/10/31/misp-dashboard-real-time-visualization-misp-events/), php-redis is present (v. 3.1.2-1). Am I missing something? :confused: 
Obviously, the [IP]:8001 web page shows me empty maps.

**Comment by mokaddem** [comment]
Hi,
They are requirements for the plugin to work:
```
# MISP has a new pub/sub feature, using ZeroMQ. To enable it, simply run the following command
sudo pip install pyzmq
# ZeroMQ depends on the Python client for Redis
sudo pip install redis
```
Did you install them on the VM side? If I'm not mistaken, they are not installed by default in the VM.

**Comment by P4rs3R** [comment]
@mokaddem yep, I installed all the stuff - as I wrote at the beginning, I'm running zmq 17.0.0 / 4.1.6, otherwise I'd have error while importing the library. The same is for redis (and redis-server v. 4.0.1 seems running properly).
PS: don't know if somehow related (probably not), but sometimes the MISP web-page serverSettings/diagnostics shows me only "An Internal Error Has Occurred."

**Comment by mokaddem** [comment]
Concerning your 'PS' I don't know if it is related...
However, have you tried stoping and starting ZeroMQ again in the ``Server Settings & Maintenance -> Diagnostics`` page? 
Only restarting the VM might not be sufficient.
If yes, what does the status says?

**Comment by iglocska** [comment]
When you get the internal error, could you check
/var/www/MISP/app/tmp/logs/error.log right after that?

**Comment by P4rs3R** [comment]
Hi guys, don't know how, but just after some apt-get dist-upgrade (and similar stuff) I managed to run MISP diagnostic, logout, ... (and ZeroMQ!). Most of all, maybe I've had some issues with my network trafic (next time, I'll try everything just in local). Now everything works, but I'm looking for some redis settings, to save events to a temporary db, instead of using RAM, as I'm saying [here](https://github.com/MISP/misp-dashboard/pull/43#issuecomment-373671848). Thank you and ... my compliment, indeed.

**Comment by scsich** [comment]
I had a similar issue here after following the install guides here:

https://misp.github.io/MISP/INSTALL.ubuntu1804/

For me the solution turned out to be installing system wide zmq and redis python3 packages:

`apt install python3-redis python3-zmq`

The virtual environment for the modules has these which was misleading to me at least. @P4rs3R , did you happen to be checking for these within the misp-modules virtual env and not system wide?
